### PR TITLE
Add readme for testing controller

### PR DESCRIPTION
**Pull request by richardwxn**

### CODE REVIEW

File: pkg/controller/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+### Controller local test
+#### Run inside the cluster
+1.run make docker.all to push image with your HUB and TAG
+
+2.update deploy/operator.yaml to point to your image
+
+3.run kubectl apply --recursive deploy/
```
Comment (sdake) [inline pkg/controller/README.md]: this is wrong.  instead use `kubectl apply -k deploy/` followed by an application of the CRD file.

File: pkg/controller/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+### Controller local test
+#### Run inside the cluster
+1.run make docker.all to push image with your HUB and TAG
+
+2.update deploy/operator.yaml to point to your image
```
Comment (sdake) [inline pkg/controller/README.md]: md formatting requires a 1 here.

File: pkg/controller/README.md (hunk 1 above)
Comment (sdake) [reply pkg/controller/README.md]: md formatting requires a 1 here.

File: pkg/controller/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+### Controller local test
```
Comment (sdake) [inline pkg/controller/README.md]: should be 1 or 2 `#` not 3.

File: pkg/controller/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#### Run inside the cluster
```
Comment (sdake) [inline pkg/controller/README.md]: should be 2 or 3 `#` not 4.

File: pkg/controller/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+1.run make docker.all to push image with your HUB and TAG
+
+2.update deploy/operator.yaml to point to your image
+
+3.run kubectl apply --recursive deploy/
+
+#### Run outside the cluster
```
Comment (sdake) [inline pkg/controller/README.md]: should be 2 or 3 `# not 4.

File: pkg/controller/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+### Controller local test
+#### Run inside the cluster
+1.run make docker.all to push image with your HUB and TAG
```
Comment (linsun) [inline pkg/controller/README.md]: can we add a step on specify HUB and TAG first?

**Comment by linsun** [comment]
Thanks for adding this @richardwxn

File: pkg/controller/README.md (hunk 6 above)
Comment (linsun) [reply pkg/controller/README.md]: also whether user needs to export BUILD_WITH_CONTAINER

File: pkg/controller/README.md (hunk 3 above)
Comment (richardwxn) [reply pkg/controller/README.md]: done

File: pkg/controller/README.md (hunk 4 above)
Comment (richardwxn) [reply pkg/controller/README.md]: done

File: pkg/controller/README.md (hunk 5 above)
Comment (richardwxn) [reply pkg/controller/README.md]: done

File: pkg/controller/README.md (hunk 6 above)
Comment (richardwxn) [reply pkg/controller/README.md]: ok, done

File: pkg/controller/README.md (hunk 2 above)
Comment (richardwxn) [reply pkg/controller/README.md]: done

File: pkg/controller/README.md (hunk 6 above)
Comment (sdake) [reply pkg/controller/README.md]: BUILD_WITH_CONTAINER is default.  I am not sure HUB nor TAG are passed into the make core file.  I will investigate this and fix it if it is busted.

Comment (richardwxn) [reply pkg/controller/README.md]: @sdake yes actually I did not use BUILD_WITH_CONTAINER for this step, but looks like you already added passing HUB and TAG in common files. 

Also another reason I failed to run make docker.all with container was some docker permission issue, but other make target works fine.

**Comment by ostromart** [comment]
@sdake and @linsun could we get this merged? We have a volunteer tester (@nrjpoddar ) who will try out these docs.

**Comment by ostromart** [comment]
Approving so @nrjpoddar can try it, please open an issue if anything still needs updating.

**Review by ostromart — APPROVED** [review]

**Pull request merged**